src/main/java/com/baikalplatform/azure/MetricsRetriever.java

```diff
@@ -81,19 +82,25 @@ public void run() {
 
         var metrics = new HashMap<String, Integer>();
 
-        var success = response.statusCode() == 200;
-        if (success) {
+        if (response.statusCode() == HttpURLConnection.HTTP_OK) {
             var optionalHeader = response.headers().firstValue(AZURE_HEADER_RATELIMIT_REMAINING);
 
             optionalHeader.ifPresent(header -> {
-                logger.info("Health probe OK: {}", header);
+                logger.info("Health probe OK (200): {}", header);
                 var elements = header.split(",");
                 Stream.of(elements).forEach(s -> {
                     var values = s.split(";");
                     metrics.put(values[0], Integer.parseInt(values[1]));
                 });
             });
+        } else if (response.statusCode() == 429) { // too many requests, not available in HttpURLConnection
+            var header = response.headers()
+                                 .firstValue(AZURE_HEADER_RATELIMIT_REMAINING)
+                                 .orElseGet(() -> "no header " + AZURE_HEADER_RATELIMIT_REMAINING);
+            logger.info("Health probe TOO_MANY_REQUESTS (429): {}", header);
+            metrics.clear();
```

Comment:
Once I reproduce this scenario we'll see if the header is also present and we can decide whether it's better to set metrics' value to zero or `clear` them.